### PR TITLE
#814: Add child areas to area landing page

### PR DESCRIPTION
**Pull request by joegasewicz**
### What

Describe what you have changed and why.

### How to review

Describe the steps required to test the changes.

### Who can review

Describe who worked on the changes, so that other people can review.

### CODE REVIEW

File: areas/areas_test.go [hunk 1]
```diff
@@ -243,6 +243,50 @@ func TestClient_GetArea(t *testing.T) {
...
+			},
+			{
+				"area_code": "E12000003",
+				"area_name": "Yorkshire and The Humbe",
+				"href": "/v1/area/E12000003"
+			}
+		]`
+	acceptedLang := "en-GB,en-US;q=0.9,en;q=0.8"
+	Convey("When a bad request is returned", t, func() {
+		mockedApi := getMockAreaAPI(http.Request{Method: "GET"}, MockedHTTPResponse{StatusCode: 400, Body: ""})
```
Comment (mike-kelly-methods) [inline areas/areas_test.go:267]: Should be using `http.StatusBadRequest` here instead of `400`

File: areas/areas_test.go [hunk 2]
```diff
@@ -243,6 +243,50 @@ func TestClient_GetArea(t *testing.T) {
...
+		_, err := mockedApi.GetRelations(ctx, userAuthToken, serviceAuthToken, collectionID, "E92000001", acceptedLang)
+		So(err, ShouldNotBeNil)
+	})
+
+	Convey("When relations are returned", t, func() {
+		mockedApi := getMockAreaAPI(http.Request{Method: "GET"}, MockedHTTPResponse{StatusCode: 200, Body: relationsBody})
```
Comment (mike-kelly-methods) [inline areas/areas_test.go:273]: Should be using `http.StatusOK` here instead of `200`

File: areas/areas_test.go [hunk 3]
```diff
@@ -243,6 +243,50 @@ func TestClient_GetArea(t *testing.T) {
...
+		relations, err := mockedApi.GetRelations(ctx, userAuthToken, serviceAuthToken, collectionID, "E92000001", acceptedLang)
+		So(err, ShouldBeNil)
+		So(relations, ShouldResemble, relations)
+	})
+	Convey("given a 200 status with valid empty body is returned", t, func() {
+		mockedApi := getMockAreaAPI(http.Request{Method: "GET"}, MockedHTTPResponse{StatusCode: 200, Body: "[]"})
```
Comment (mike-kelly-methods) [inline areas/areas_test.go:279]: Ditto above

File: areas/areas.go [hunk 4]
```diff
@@ -105,6 +103,34 @@ func (c *Client) GetArea(ctx context.Context, userAuthToken, serviceAuthToken, c
...
+		err = NewAreaAPIResponse(res, uri)
+		return
+	}
+
+	b, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return
+	}
+	var body []Relation
+	if err = json.Unmarshal(b, &body); err != nil {
```
Comment (mike-kelly-methods) [inline areas/areas.go:127]: Is unmarshalling here necessary as you're unmarshalling the response body below too?

File: areas/areas.go [hunk 5]
```diff
@@ -105,6 +103,34 @@ func (c *Client) GetArea(ctx context.Context, userAuthToken, serviceAuthToken, c
 	return
 }
 
+// GetRelations gets the child areas
+func (c *Client) GetRelations(ctx context.Context, userAuthToken, serviceAuthToken, collectionID, areaID, acceptLang string) (relations []Relation, err error) {
+	uri := fmt.Sprintf("%s/v1/areas/%s/relations", c.hcCli.URL, areaID)
+	clientlog.Do(ctx, "retrieving area relations", service, uri)
+	// Do request
+	res, err := c.doGetWithAuthHeaders(ctx, userAuthToken, serviceAuthToken, collectionID, uri, nil, "", "en")
```
Comment (mike-kelly-methods) [inline areas/areas.go:111]: Rather that inline the string `"en"` can this be a constant and reference that?

File: areas/areas.go (hunk 4 above)
Comment (joegasewicz) [reply areas/areas.go:127]: Sorry, my bad

**Review by mike-kelly-methods — APPROVED** [review]
👍

**Pull request closed without merging**